pallets/api/src/fungibles/mod.rs

```diff
@@ -460,13 +462,17 @@ pub mod pallet {
 		/// - `account` - The account from which the tokens will be destroyed.
 		/// - `value` - The number of tokens to destroy.
 		#[pallet::call_index(20)]
-		#[pallet::weight(AssetsWeightInfoOf::<T>::burn())]
+		#[pallet::weight(<T as Config>::WeightInfo::balance_of() + AssetsWeightInfoOf::<T>::burn())]
 		pub fn burn(
 			origin: OriginFor<T>,
 			token: TokenIdOf<T>,
 			account: AccountIdOf<T>,
 			value: BalanceOf<T>,
 		) -> DispatchResult {
+			let current_balance = AssetsOf::<T>::balance(token.clone(), &account);
+			if current_balance < value {
+				return Err(AssetsErrorOf::<T>::BalanceLow.into());
+			}
```

Comment:
It should only return the weight of querying the balance if it errors

Reply:
Resolved!